src/coreclr/interpreter/compiler.cpp

```diff
@@ -2189,6 +2218,13 @@ void InterpCompiler::EmitLoadVar(int32_t var)
 
 void InterpCompiler::EmitStoreVar(int32_t var)
 {
+    // Check for the unusual case of storing to the this pointer variable within a generic method which uses the this pointer as a generic context arg
+    // and if doing so and the mitigation for this pointer changing is not enabled, restart the compile to enable it
```

Comment:
Comment mentions restarting compilation but it looks like we don't do that anymore